miio/gateway.py

```diff
@@ -536,38 +513,216 @@ class SubDevice:
     def __init__(self, gw, sid, type_, _, __, ___):
         self.gw = gw
         self.sid = sid
-        self.type = DeviceType(type_)
+        self._battery = None
+        try:
+            self.type = DeviceType(type_)
+        except ValueError:
+            self.type = DeviceType(-1)
 
-    def unpair(self):
-        return self.gw.send("remove_device", [self.sid])
+    def __repr__(self):
+        return "<Subdevice %s: %s fw: %s bat: %s>" % (
+            self.device_type,
+            self.sid,
+            self.get_firmware_version(),
+            self.get_battery(),
+        )
 
+    @property
+    def device_type(self):
+        return self.type.name
+
+    @property
     def battery(self):
-        return self.gw.send("get_battery", [self.sid])[0]
+        return self._battery
 
-    def get_firmware_version(self) -> Optional[int]:
-        """Returns firmware version"""
+    @command()
+    def subdevice_send(self, command):
+        """Send a command/query to the subdevice"""
+        try:
+            return self.gw.send(command, [self.sid])
+        except Exception as ex:
```

Comment:
It is better to let this raise an exception and let the downstream to handle it however they prefer.

Reply:
done